test/util/parser_test.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2017 XLAB, Ltd.
+ *
+ * This work is open source software, licensed under the terms of the
+ * BSD license as described in the LICENSE file in the top-level directory.
+ */
+
+package util_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/mikelangelo-project/capstan/util"
+	. "gopkg.in/check.v1"
+)
+
+func Test(t *testing.T) { TestingT(t) }
+
+type testingParserSuite struct{}
+
+var _ = Suite(&testingParserSuite{})
+
+func (s *testingParserSuite) TestParseEnvironmentList(c *C) {
+	m := []struct {
+		comment     string
+		envList     []string
+		expectedRes []string
+		err         string
+	}{
+		{
+			"no parameters",
+			[]string{},
+			[]string{},
+			"",
+		},
+		{
+			"single parameter",
+			[]string{"PORT=8000"},
+			[]string{"PORT=>8000"},
+			"",
+		},
+		{
+			"two parameters",
+			[]string{"PORT=8000", "ENDPOINT=foo.com"},
+			[]string{"PORT=>8000", "ENDPOINT=>foo.com"},
+			"",
+		},
+		{
+			"invalid char (space) #1",
+			[]string{"NAME=my name"},
+			[]string{},
+			"failed to parse --env argument .*",
+		},
+		{
+			"invalid char (space) #2",
+			[]string{"MY NAME=name"},
+			[]string{},
+			"failed to parse --env argument .*",
+		},
+		{
+			"invalid char (space) #3",
+			[]string{"MY NAME=my name"},
+			[]string{},
+			"failed to parse --env argument .*",
+		},
+		{
+			"invalid char (equals sign) #1",
+			[]string{"NAME=my=name"},
```

Comment:
This is not actually a problem for OSv. It works the same as `putenv` or BASH:

```bash
lBook-2:~ lemmy$ export VAR=val_var=value
lBook-2:~ lemmy$ echo $VAR
val_var=value
```

I suggest we allow such configuration. Namely, the first `=` splits between variable name and value. Value itself can then contain `=`.